flytekit/clis/flyte_cli/main.py

```diff
@@ -1478,64 +1482,69 @@ def register_project(identifier, name, description, host, insecure):
-            f"Resource type found in identifier {id.resource_type} invalid, must be launch plan, " f"task, or workflow"
+            f"Resource type found in identifier {resource_type_name} invalid, must be launch plan, task, or workflow"
         )
-    entity = _load_proto_from_file(resource_map[id.resource_type], object_file)
-    return id, entity
+
+    entity = _load_proto_from_file(pb2_type, proto_file)
+    return hydrate_registration_parameters(resource_type, project, domain, version, entity)
 
 
-def _extract_files(file_paths):
+def _extract_files(project: str, domain: str, version: str, file_paths: List[str]):
     """
     :param file_paths:
     :rtype: List[(flyteidl.core.identifier_pb2.Identifier, T)]
     """
-    # Get a manual iterator because we're going to grab files two at a time.
-    # The identifier file will always come first because the names are always the same and .identifier.pb sorts before
-    # .pb
-
     results = []
     filename_iterator = iter(file_paths)
-    for identifier_file in filename_iterator:
-        object_file = next(filename_iterator)
-        id, entity = _extract_pair(identifier_file, object_file)
-        results.append((id, entity))
+    for proto_file in filename_iterator:
+        # Serialized proto files are of the form: foo.bar.<resource_type>.pb
+        file_name_parts = proto_file.split(".")
+        if file_name_parts[-1] != "pb":
+            _click.echo(f"Skipping non-proto file {proto_file}")
+            continue
+        if len(file_name_parts) <= 2:
+            raise Exception(f"Serialized proto file {proto_file} has unrecognized file name")
+        resource_type_name = file_name_parts[-2]
+        results.append(_get_entity(project, domain, version, resource_type_name, proto_file))
 
     return results
 
 
 @_flyte_cli.command("register-files", cls=_FlyteSubCommand)
+@_click.option(*_PROJECT_FLAGS, required=True, help="The project namespace to register with.")
+@_click.option(*_DOMAIN_FLAGS, required=True, help="The domain namespace to register with.")
+@_click.option(*_VERSION_FLAGS, required=True, help="The entity version to register with")
 @_host_option
 @_insecure_option
 @_click.argument(
     "files", type=_click.Path(exists=True), nargs=-1,
 )
-def register_files(host, insecure, files):
+def register_files(project, domain, version, host, insecure, files):
     """
     Given a list of files, this will (after sorting the input list), attempt to register them against Flyte Admin.
     This command expects the files to be the output of the pyflyte serialize command.  See the code there for more
-    information. Valid files need to be:
+    information. Valid files need to be:\n
         * Ordered in the order that you want registration to happen. pyflyte should have done the topological sort
-          for you and produced file that have a prefix that sets the correct order.
+          for you and produced file that have a prefix that sets the correct order.\n
```

Comment:
do we really need these?

Reply:
don't we still need the topological sort to guarantee that we register tasks before workflows that reference them, and workflows before launch plans that reference them?